### PR TITLE
Fix warns occuring in C++ tests

### PR DESCRIPTION
**Pull request by FelonEkonom**

### CODE REVIEW

File: test/unifex/integration_test.exs [hunk 1]
```diff
@@ -71,6 +71,10 @@ defmodule Unifex.IntegrationTest do
     "test/fixtures/#{project}_ref_generated/#{interface}"
     |> File.ls!()
     |> Enum.each(fn ref ->
+      # File.read!("test_projects/#{project}/c_src/example/_generated/#{interface}/#{ref}")
+      # |> IO.puts()
+      # IO.puts("dupa #{project} #{interface} #{ref}")
+
```
Comment (mat-hek) [inline test/unifex/integration_test.exs:77]:
```suggestion
```

File: test_projects/cnode/bundlex.exs [hunk 2]
```diff
@@ -8,13 +8,14 @@ defmodule Example.BundlexProject do
   end
 
   def natives(_platform) do
+    language = System.get_env("UNIFEX_TEST_LANG") |> String.to_atom()
     [
       example: [
         src_base: "example",
-        sources: ["example.c"],
+        sources: ["example.#{language}"],
```
Comment (mat-hek) [inline test_projects/cnode/bundlex.exs:15]: Do C and C++ files differ from each other? If not, I'd try not to duplicate their contents

**Review by mat-hek — CHANGES_REQUESTED** [review]

File: test_projects/cnode/bundlex.exs (hunk 2 above)
Comment (FelonEkonom) [reply test_projects/cnode/bundlex.exs:15]: There is no difference between these two files, but C++ compiler generates warnings on try of compiling a file with `.c` extension

Comment (mat-hek) [reply test_projects/cnode/bundlex.exs:15]: So maybe we could create the cpp files dynamically in test?

File: test/unifex/integration_test.exs [hunk 3]
```diff
@@ -2,33 +2,57 @@ defmodule Unifex.IntegrationTest do
   use ExUnit.Case, async: true
 
   test "NIF test project" do
+    generate_cpp_code("nif")
+
     test_project("nif", :nif, :c)
     test_project("nif", :nif, :cpp)
+
+    delete_cpp_code("nif")
```
Comment (mat-hek) [inline test/unifex/integration_test.exs:10]: Let's rename `test_project` to something like `do_test_project` and move this to a new `test_project` function. Also, delete_cpp_code should be run with `on_exit` https://hexdocs.pm/ex_unit/ExUnit.Callbacks.html#on_exit/2

**Review by mat-hek — CHANGES_REQUESTED** [review]

File: test/unifex/integration_test.exs (hunk 3 above)
Comment (FelonEkonom) [reply test/unifex/integration_test.exs:10]: Problem is, that `on_exit` should be called from `setup` function, which is shared between many test functions and each test generates and deletes files from different directory

Comment (mat-hek) [reply test/unifex/integration_test.exs]: `on_exit` can be called from within a test as well. If you don't use it, the files won't be deleted when the test fails

File: test/unifex/integration_test.exs [hunk 4]
```diff
@@ -75,4 +79,12 @@ defmodule Unifex.IntegrationTest do
                File.read!("test/fixtures/#{project}_ref_generated/#{interface}/#{ref}")
     end)
   end
+
+  defp generate_cpp_code(project) do
+    System.cmd("cp", ["example.c", "example.cpp"], cd: "test_projects/#{project}/c_src/example")
```
Comment (mat-hek) [inline test/unifex/integration_test.exs:84]: Actually, you can use `File.copy!/2`

File: test/unifex/integration_test.exs [hunk 5]
```diff
@@ -75,4 +79,12 @@ defmodule Unifex.IntegrationTest do
...
+  end
+
+  defp delete_cpp_code(project) do
+    System.cmd("rm", ["example.cpp"], cd: "test_projects/#{project}/c_src/example")
```
Comment (mat-hek) [inline test/unifex/integration_test.exs:88]: `File.rm`

File: test/unifex/integration_test.exs (hunk 3 above)
Comment (mat-hek) [reply test/unifex/integration_test.exs]: bump

File: test/unifex/integration_test.exs [hunk 6]
```diff
@@ -2,60 +2,40 @@ defmodule Unifex.IntegrationTest do
...
+    generate_cpp_code(project)
 
-    test_project("bundlex_exs", :cnode, :c)
-    test_project("bundlex_exs", :cnode, :cpp)
+    for language <- [:c, :cpp] do
+      do_test_project(project, interface, language)
+    end
 
-    delete_cpp_code("bundlex_exs")
+    delete_cpp_code(project)
```
Comment (mat-hek) [inline test/unifex/integration_test.exs:35]: ON_EXIT

**Review by mat-hek — CHANGES_REQUESTED** [review]

**Review by mat-hek — APPROVED** [review]

**Pull request merged**